src/garage/envs/environment.py

```diff
@@ -0,0 +1,188 @@
+"""Base Garage Environment API."""
+
+import abc
+
+
+class Environment(abc.ABC):
+    """The main API for garage environments.
+
+    The main public API methods are:
+        * step()
+        * reset()
+        * render()
+        * visualize()
+        * close()
+
+    Set the following attributes:
+        * action_space: The action space specification
+        * observation_space: The observation space specification
+        * spec: The environment specifications
+        * render_modes: The list of supported render modes
+
+    Make sure your environment is pickle-able:
```

Comment:
does this actually render well in the docs? i think you might need to adjust the markup here.

typically in Python docs this kind of info is presented as an admonition. Example: https://recommonmark.readthedocs.io/en/latest/auto_structify.html#recommonmark.transform.AutoStructify